src/genomicsdb.pyx

```diff
@@ -305,6 +312,78 @@ cdef class _GenomicsDB:
                                              as_ranges(row_ranges))
       
       return pandas.DataFrame(processor.construct_data_frame()).replace(np.nan, '').replace(-99999, '');
+
+    def query_variant_calls_arrow(self,
+                                  array=None,
+                                  column_ranges=None,
+                                  row_ranges=None,
+                                  query_protobuf: query_pb.QueryConfiguration=None,
+                                  non_blocking=False): 
+      """ Query for variant calls from the GenomicsDB workspace using array, column_ranges and row_ranges for subsetting """
+
+      cdef ArrowVariantCallProcessor processor
+      cdef void *arrow_array = NULL
+      cdef void *arrow_schema = NULL
+
+      if non_blocking:
+        processor.set_threaded(1)
+
+      def query_calls():
+        if query_protobuf:
+          if array or column_ranges or row_ranges:
+            raise GenomicsDBException("Cannot specify query_protobuf and array/column_ranges/row_ranges together")
+          configstring = as_protobuf_string(query_protobuf.SerializeToString())
+          with nogil:
+            self._genomicsdb.query_variant_calls(processor, configstring, GENOMICSDB_PROTOBUF_BINARY_STRING)
+        elif array is None:
+          configstring = as_string("")
+          with nogil:
+            self._genomicsdb.query_variant_calls(processor, configstring, GENOMICSDB_NONE)
+        elif column_ranges is None:
+          configstring = as_string(array)
+          rows = scan_full()
+          with nogil:
+            self._genomicsdb.query_variant_calls(processor, configstring, rows)
+        elif row_ranges is None:
+          configstring = as_string(array)
+          columns = as_ranges(column_ranges)
+          with nogil:
+            self._genomicsdb.query_variant_calls(processor, configstring, columns)
+        else:
+          configstring = as_string(array)
+          columns = as_ranges(column_ranges)
+          rows = as_ranges(row_ranges)
+          with nogil:
+            self._genomicsdb.query_variant_calls(processor, configstring,
+                                                 columns, rows)
+
+      if non_blocking:
+        query_thread = threading.Thread(target=query_calls)
+        query_thread.start()
+      else:
+        query_calls()
+
+      cdef object schema_capsule
+      while True:
+        arrow_array = processor.arrow_array()
+        if arrow_schema == NULL:
+          arrow_schema = processor.arrow_schema();
+          schema_capsule = pycapsule_get_arrow_schema(arrow_schema)
+          schema_obj = _ArrowSchemaWrapper._import_from_c_capsule(schema_capsule)
+          schema = pa.schema(schema_obj.children_schema)
+          yield schema.serialize().to_pybytes()
+        if arrow_array and arrow_schema:
```

Comment:
The `and arrow_schema` part of this check seems redundant given the `arrow_schema == NULL` check just above, or is there some truthiness conversion?

Reply:
Agreed!